openedx/features/genplus_features/genplus/api/v1/views.py

```diff
@@ -175,45 +182,70 @@ def get_serializer_class(self):
+        if skill_id:
+            queryset = queryset.filter(skill__id=skill_id)
+        queryset = queryset.filter(Q(title__icontains=search) | Q(description__icontains=search))
+        return queryset
+
+    def _booster_badge_filter(self, queryset, request, *args, **kwargs):
+        query_params = self.request.query_params
+        skill_id = query_params.get('skill_id')
+        if skill_id:
+            return BoosterBadgeAward.objects.none()
+        search = query_params.get('search', '')
+        queryset = queryset.filter(feedback__icontains=search)
+        return queryset
+
+    def _rename_entry_type(self, results):
+        for i, entry in enumerate(results):
+            if entry['type'] == 'JournalPost':
+                results[i]['type'] = results[i]['journal_type']
+        return results
+
+    def get_querylist(self):
         query_params = self.request.query_params
         journal_posts = JournalPost.objects.select_related('student', 'teacher', 'skill')
+
         if self.gen_user.is_student:
-            journal_posts = journal_posts.filter(student=self.gen_user.student)
+            student = self.gen_user.student
+            journal_posts = journal_posts.filter(student=student)
+            booster_badges = BoosterBadgeAward.objects.filter(user=self.gen_user.user)
         else:
             student_id = query_params.get('student_id')
-            if student_id:
-                journal_posts = journal_posts.filter(student__id=student_id)
-            else:
-                journal_posts = JournalPost.objects.none()
-
-        skill = query_params.get('skill')
-        if skill:
-            journal_posts = journal_posts.filter(skill__name__iexact=skill)
-
-        return journal_posts.order_by(*self.ordering)
+            student = get_object_or_404(Student, pk=student_id)
+            journal_posts = journal_posts.filter(student=student)
+            booster_badges = BoosterBadgeAward.objects.filter(user__gen_user=student.gen_user)
+
+        return [
+            {
+                'queryset': journal_posts,
+                'serializer_class': JournalListSerializer,
+                'filter_fn': self._journal_entry_filter
+            },
+            {
+                'queryset': booster_badges,
+                'serializer_class': JournalBoosterBadgeSerializer,
+                'filter_fn': self._booster_badge_filter
+            }
+        ]
 
     def list(self, request, *args, **kwargs):
-        queryset = self.filter_queryset(self.get_queryset())
-        skill_qs = Skill.objects.all()
-        skills = SkillSerializer(skill_qs, many=True).data
-        page = self.paginate_queryset(queryset)
-
-        if page is not None:
-            data = self.get_serializer(page, many=True).data
-            response = self.get_paginated_response(data)
-            response.data['skills'] = skills
-            return response
+        response = super(FlatMultipleModelMixin, self).list(request, args, kwargs)
```

Comment:
removed skills from the queryset, since there is a separate endpoint for skills.